### ISSUE TITLE
Code clarification

### ISSUE THREAD

**Issue by kamalkraj** [zalandoresearch/flair#55]
https://github.com/zalandoresearch/flair/blob/3635220bdc05c19e4fe47555071db6c39c009bae/flair/models/language_model.py#L41
What is `nout` ?
In the Tutorial for  your own Character LM Embeddings also didn't mention about the parameter `nout`
Can you also give the original configuration used to train 1-billion words

**Comment by alanakbik** [comment]
Hi Kamal, thanks for your interest! 

`nout` is an optional linear layer after the RNN that will downproject to a smaller space before making character predictions. This can allow you to have a large language model - let's say one with 2048 states - but make predictions based on a smaller space and thus product smaller embeddings. So, you could have a huge LM that creates embeddings of dimensionality 256 or so. However, in our experiments so far, it has always been better to skip the downprojection and just use the full language model. 

The parameters are as follows: 
- 2048 hidden states (1024 for the 'fast' variant)
- 1 layer
- sequence length of 250
- mini batch size of 100 
- no downprojection (`nout` = None)
- initial learning rate of 20, annealing factor of 25
- anneal patience of 25 (i.e. anneal after 25 1B word corpus splits of no dev perplexity improvement)

Hope this helps!
